pkg/controller/nebulascheduledbackup/nebula_scheduled_backup_manager.go

```diff
@@ -0,0 +1,396 @@
+		klog.Infof("Most recent backup job %s/%s for Nebula scheduled backup %s/%s succeeded", lastSuccessfulBackup.Namespace, lastSuccessfulBackup.Name, scheduledBackup.Namespace, scheduledBackup.Name)
+	} else if len(failedBackups) != 0 {
+		lastFailedBackup := failedBackups[len(failedBackups)-1]
+		lastBackupFailed = true
+		klog.Errorf("Most recent backup job %s/%s for Nebula scheduled backup %s/%s failed", lastFailedBackup.Namespace, lastFailedBackup.Name, scheduledBackup.Namespace, scheduledBackup.Name)
+	} else {
+		klog.Info("No previous successful or failed backup jobs detected.")
+	}
+
+	// Check Nebula Graph cluster status and return error if it's down.
+	ns := scheduledBackup.GetNamespace()
+	if scheduledBackup.Spec.BackupTemplate.BR.ClusterNamespace != nil {
+		ns = *scheduledBackup.Spec.BackupTemplate.BR.ClusterNamespace
+	}
+
+	nc, err := bm.clientSet.NebulaCluster().GetNebulaCluster(ns, scheduledBackup.Spec.BackupTemplate.BR.ClusterName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get nebula cluster %s/%s err: %w", ns, scheduledBackup.Spec.BackupTemplate.BR.ClusterName, err)
+	}
+
+	if !nc.IsReady() {
+		return nil, nil, fmt.Errorf("nebula cluster %s/%s is not ready", ns, scheduledBackup.Spec.BackupTemplate.BR.ClusterName)
+	}
+
+	// Calculate previous and next backup time
+	previousBackupTime, nextBackupTime, err := calculateRunTimes(scheduledBackup, now)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to calculate next backup time for Nebula scheduled backup %s/%s. Err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+	}
+
+	if previousBackupTime != nil && now.After(previousBackupTime.Local()) && (scheduledBackup.Status.CurrPauseStatus == nil || !*scheduledBackup.Status.CurrPauseStatus) {
+		if len(runningBackups) > 0 {
+			// active running backup from last iteration has not completed. Issue warning and do not start new backup
+			klog.Warningf("Active backup job from last scheduled run detected for Nebula scheduled backup %s/%s. Will not kick off new backup job. Please consider decreasing the backup frequency by adjusting the schedule.", scheduledBackup.Namespace, scheduledBackup.Name)
+			if len(runningBackups) != 1 {
+				// This could be because someone manually started a nebula backup and associated it to this scheduled backup. Need to issue another warning.
+				klog.Warningf("More than 1 active backup detected for Nebula scheduled backup %s/%s. Please check if the second job was manually started and associated to this crontab", scheduledBackup.Namespace, scheduledBackup.Name)
+			}
+
+			for _, backup := range runningBackups {
+				klog.Warningf("Active backup job %v detected", backup.Name)
+			}
+			return &kube.ScheduledBackupUpdateStatus{
+				CurrPauseStatus:          scheduledBackup.Spec.Pause,
+				LastSuccessfulBackupTime: lastSuccessfulTime,
+				MostRecentJobFailed:      &lastBackupFailed,
+			}, nextBackupTime, nil
+		} else {
```

Comment:
else if redundancy

Reply:
I think you meant else has redundant code. Fixed by combining the return statement into 1 at the end of the function.